### PR TITLE
Switch to jQuery 2

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Switches jQuery from `1.x` to `2.x`. The previous version was mostly designed to maintain browser compatibility for browsers that were considered old at the time... which was about 9 years ago!

The performance should be better, especially in regards to javascript memory leaks...

#### What should we test?

Maybe have a quick navigate around the app and look for new console errors? In theory a green build should tell us if anything is broken.

#### Release notes

Changelog Category: Technical changes

### CODE REVIEW

**Comment by jibees** [comment]
Did you consider removing jQuery?

**Comment by Matt-Yorkley** [comment]
> Did you consider removing jQuery?

That should be our goal eventually, but it'll take a lot more than a few lines of code :sweat_smile:

**Comment by jibees** [comment]
Yeap, sure. Maybe my next task with "tech debt" label?

File: app/assets/javascripts/darkswarm/all.js.coffee [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#= require jquery
+#= require jquery2
```
Comment (jibees) [inline app/assets/javascripts/darkswarm/all.js.coffee:1]: Just to be sure to understand. Why not using yarn (and `package.json`) for this one?

File: app/assets/javascripts/darkswarm/all.js.coffee [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
 #= require jquery_ujs
 #= require jquery.ui.all
```
Comment (jibees) [inline app/assets/javascripts/darkswarm/all.js.coffee:3]: Does this one relies on jQuery 1 or 2 or whatever?

File: app/assets/javascripts/darkswarm/all.js.coffee (hunk 1 above)
Comment (Matt-Yorkley) [reply app/assets/javascripts/darkswarm/all.js.coffee:1]: So this is currently provided by a gem (jquery-rails) and that gem contains different versions of the jquery code which is sightly customised. At some point we can look at moving all javascript assets to being imported and installed via yarn and webpacker, but this was the simplest and quickest way to switch to `2.x`.

File: app/assets/javascripts/darkswarm/all.js.coffee (hunk 2 above)
Comment (Matt-Yorkley) [reply app/assets/javascripts/darkswarm/all.js.coffee:3]: They should be compatible :+1:

File: app/assets/javascripts/darkswarm/all.js.coffee (hunk 1 above)
Comment (jibees) [reply app/assets/javascripts/darkswarm/all.js.coffee:1]: Thanks!

File: app/assets/javascripts/darkswarm/all.js.coffee (hunk 2 above)
Comment (jibees) [reply app/assets/javascripts/darkswarm/all.js.coffee:3]: Nice! 👌

**Review by jibees — APPROVED** [review]
👌

**Review by mkllnk — APPROVED** [review]
Nice move. :+1:

I would feel more confident if we had fixed and merged #7393 but I guess that we will notice if things break...

**Comment by RachL** [comment]
I don't think I've stumbled upon new errors in the console. On staging fr there was the error regarding pictures, which is knows and the following warnings:

image

i think we are good to go :+1:

**Pull request merged**